### DISCUSSION TITLE
Monitor of EasyEffects Sink vs EasyEffects Source

### DISCUSSION THREAD

**Discussion by JaakkoHintsala** [wwmm/easyeffects#1391] [General]
At least for zoom launched in firefox I don't see any difference in choosing either one of these as microphone. 

Isn't sink supposed to be an output and source an input? 

Could someone elaborate what is the difference between these two?

**Comment by wwmm** [comment]
>At least for zoom launched in firefox I don't see any difference in choosing either one of these as microphone.
>Isn't sink supposed to be an output and source an input?

That is because Zoom is deciding to show monitors as well as standard sources. Even your sound card sink has monitors that can be used as sources. That is normal. It is up to the applications if they will use them or not.

`EasyEffects Sink` is the virtual device applications have to play to when we want to add effects to their output. As any other sink it also has monitors you could use to process the signal the sink is receiving. The input of our filters pipeline is linked to these monitors. That is how we are able to process applications output.

`EasyEffects Source` has a similar role but in this case it is the output of our filters pipeline that is linked to it. And audio record applications will be the one recording from this virtual device.

>Could someone elaborate what is the difference between these two?

If you want to record a microphone signal that has been processed by EasyEffects you have to use `EasyEffects Source` as source. And if you want effects to be applied to an app output you have to make them use `EasyEffects Sink` as output.
